From 85194e6fa201df6c39edef86b2965e0d86af3528 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:20:27 -0400
Subject: [fs] nfs: fix comment and add warn_on for PG_INODE_REF

Message-id: <1411050057-12417-65-git-send-email-steved@redhat.com>
Patchwork-id: 96223
O-Subject: [RHEL7.1 PATCH 64/94] nfs: fix comment and add warn_on for PG_INODE_REF
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Weston Andros Adamson <dros@primarydata.com>

Fix the comment in nfs_page.h for PG_INODE_REF to reflect that it's no longer
set only on head requests. Also add a WARN_ON_ONCE in nfs_inode_remove_request
as PG_INODE_REF should always be set.

Suggested-by: Peng Tao <tao.peng@primarydata.com>
Signed-off-by: Weston Andros Adamson <dros@primarydata.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: b412ddf0661e11485876a202c48868143e3a01cf
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/write.c b/fs/nfs/write.c
index faa3a4b..9c080f8 100644
--- a/fs/nfs/write.c
+++ b/fs/nfs/write.c
@@ -707,6 +707,8 @@ static void nfs_inode_remove_request(struct nfs_page *req)
 
  if (test_and_clear_bit(PG_INODE_REF, &req->wb_flags))
   nfs_release_request(req);
+ else
+  WARN_ON_ONCE(1);
 }
 
 static void
diff --git a/include/linux/nfs_page.h b/include/linux/nfs_page.h
index 291924c..6ad2bbc 100644
--- a/include/linux/nfs_page.h
+++ b/include/linux/nfs_page.h
@@ -26,7 +26,7 @@ enum {
  PG_MAPPED,  /* page private set for buffered io */
  PG_CLEAN,  /* write succeeded */
  PG_COMMIT_TO_DS, /* used by pnfs layouts */
- PG_INODE_REF,  /* extra ref held by inode (head req only) */
+ PG_INODE_REF,  /* extra ref held by inode when in writeback */
  PG_HEADLOCK,  /* page group lock of wb_head */
  PG_TEARDOWN,  /* page group sync for destroy */
  PG_UNLOCKPAGE,  /* page group sync bit in read path */
-- 
1.7.1